deploy/operator/deploy.sh

```diff
@@ -3,24 +3,64 @@
 set -o nounset
 set -o pipefail
 set -o errexit
-set -o xtrace
 
 __dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+__root="$(realpath ${__dir}/../..)"
 
 if [ -z "${DISKS:-}" ]; then
     export DISKS=$(echo sd{b..f})
 fi
 
+function handle_disconnected() {
+    # Some of the variables over here can be sourced from dev-scripts
+    # source common.sh utils.sh
+    # set +x
+    # export -f wrap_if_ipv6 ipversion
+
+    source ${__dir}/mirror_utils.sh
+    export MIRROR_BASE_URL="http://$(wrap_if_ipv6 ${PROVISIONING_HOST_IP})/images"
+    export AUTHFILE="${XDG_RUNTIME_DIR}/containers/auth.json"
+    mkdir -p $(dirname ${AUTHFILE})
+
+    export LOCAL_REGISTRY="${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}"
+
+    merge_authfiles "${PULL_SECRET_FILE}" "${REGISTRY_CREDS}" "${AUTHFILE}"
+}
+
+set -o xtrace
+
+#######
+# LSO #
+#######
+
 ${__dir}/libvirt_disks.sh create
 
+if [ "${DISCONNECTED:-false}" == "true" ]; then
+    handle_disconnected
+fi
+
 if [ "${INSTALL_LSO:-true}" == "true" ]; then
     ${__dir}/setup_lso.sh install_lso
 fi
 
 ${__dir}/setup_lso.sh create_local_volume
-${__dir}/setup_hive.sh with_olm
 
-export OPENSHIFT_VERSIONS="${OPENSHIFT_VERSIONS:-$(cat ${__dir}/../../data/default_ocp_versions.json)}"
+########
+# Hive #
+########
+
+if [ "${DISCONNECTED:-false}" == "true" ]; then
+  ${__root}/hack/setup_env.sh hive_from_upstream
+  ${__dir}/setup_hive.sh from_upstream
+else
+  ${__dir}/setup_hive.sh with_olm
```

Comment:
not a must, but I would also allow choosing installation method via configuration

Reply:
Would you mind to elaborate?

Reply:
``HIVE_DEPLOYMENT_METHOD`` variable which can be ``from_upstream | with_olm``